packages/solidity-contracts/test/messagesIncomingV3.test.ts

```diff
@@ -1177,6 +1179,34 @@ describe('FuelMessagePortalV3 - Incoming messages', () => {
       ).to.be.equal(true);
     });
 
+    it('Should stop applying rate limit if it is disabled', async () => {
+      await fuelMessagePortal.depositETH(messageEOA.sender, {
+        value: messageExceedingRateLimit.amount * BASE_ASSET_CONVERSION,
+      });
+
+      const [msgID, msgBlockHeader, blockInRoot, msgInBlock] = generateProof(
+        messageExceedingRateLimit,
+        blockHeaders,
+        prevBlockNodes,
+        blockIds,
+        messageNodes
+      );
+
+      await fuelMessagePortal.disableRateLimit();
```

Comment:
I'd add an additional check before disabling rate limit to verify if the relayMessage fails (when rate limit is enabled) to make sure that disabling rate limit has a real effect.